### ISSUE TITLE
Fix for initializing non-existent DOM elements

### ISSUE THREAD

**Issue by ralphjsmit** [Accudio/async-alpine#52]
Hello there! Merry Christmas and thank you for the nice package :)
I'm mostly using it indirectly in Filament context, but it's a nice and smart piece of work.

My issue and fix applies to `1.x` now as that is the one I've been testing with, but it likely applies to `2.x` also and would need to be ported there as well.

I noticed whilst working on some Filament components that when using the [Livewire SPA mode](https://livewire.laravel.com/docs/navigate), we would often be getting the following type errors in the console:

```
livewire.js?id=38dc8241:1120 Alpine Expression Error: undefined

Expression: "fileUploadFormComponent({
                    acceptedFileTypes:
                    // ... 
```

This disturbed me, because after this type of crash, it seems like other Alpine elements sometimes also did not get correctly initialized correctly and that sometimes breaks functionality.

The issue is therefore as follows. Every time the Async Alpine initializes, it does this via the `await this._generateRequirements(...)` function. Only when the requirement is met is the promise resolved and does the code flow continue to download and activate. However, because Livewire SPA moves between pages without giving a "hard refresh" to the JS, this would keep many elements still awaiting the promise from `this._generateRequirements(...)` to resolve, _whilst the element has already left the DOM_ as the user has already navigated to a next page.

```js
  async _componentStrategy(component) {
    const requirements = parseRequirements(component.strategy);

    await this._generateRequirements(component, requirements);
    await this._download(component.name);
    this._activate(component);
  },
```

If it then sometimes happens that a user has moved to a next page but the lingering promise does still resolve because e.g. an `event()` strategy requirement has been met accidentally, the code flow for these lingering promise would continue and Async Alpine would still start to initialize and activate the element, but the element is not in the DOM anymore. That then causes these issues about imported modules being undefined etc.

The best solution would be to remove these awaiting promises from the DOM on the `livewire:navigated` event, but I have no clue how to do so and if that is even possible, so the second best and also easiest solution would be to just simply `return` after the promise has resolved but if we detect that the element is not in the DOM anymore. So the above code should simply be replaced with this:

```js
async _componentStrategy(component) {
  const requirements = parseRequirements(component.strategy);

  await this._generateRequirements(component, requirements);

  if (! component.el.isConnected) {
    // component.el not present in DOM anymore (for example, by Livewire `wire:navigate`)
    return;
  }

  await this._download(component.name);

  if (! component.el.isConnected) {
    // component.el not present in DOM anymore (for example, by Livewire `wire:navigate`)
    return;
  }
  
  this._activate(component);
},
```

I tested and confirmed that this solves the issue of the promises "persisting" across the page visits. I wanted to PR this to the repo, but you don't have a `1.x` branch anymore, so I couldn't target it. I would appreciate it if you could release this fix in a `1.x` version so that it would solve the issue for all users in Filament 3.x still. It likely also needs to be ported to the `2.x` version, because that uses a same strategy of awaiting a requirements promise to resolve to true (really smart concept btw).

Thanks!

**Comment by Accudio** [comment]
Thank you for raising @ralphjsmit, good find!

I have set up a 1.x branch for critical updates to the v1 of async alpine, although I don't intent to actively maintain it. As you mention though, something like this does make sense to backport.

As I was there I made the change as you've suggested using `el.isConnected`. I've released versions [1.2.3](https://github.com/Accudio/async-alpine/releases/tag/1.2.3) and [2.0.2](https://github.com/Accudio/async-alpine/releases/tag/2.0.2) that fix this issue, thank you very much for the suggestion and filing the issue!
